lib/utils.js

```diff
@@ -483,36 +510,36 @@ module.exports = class AccessUtils {
           return actions;
         })
         .spread((currentGroupCount, targetGroupCount) => {
-          let res = false;
+          let bAllowAccess = false;
 
           if (currentGroupCount === false) {
             winston.info('No group context was determined, so allow passthrough access')
             // No group context was determined, so allow passthrough access.
-            res = true;
+            bAllowAccess = true;
           }
           else {
             // Determine grant based on the current/target group context.
-            res = currentGroupCount > 0;
+            bAllowAccess = currentGroupCount > 0;
 
-            debug(`user ${userId} ${res ? 'is a' : 'is not a'} ${roleName} of group ${scope.currentGroupId}`);
+            winston.debug(`ROLE RESOLVER - User ${userId} ${bAllowAccess ? 'has role' : 'does not have role'} ${roleName} in group ${scope.currentGroupId}`);
 
-            // If it's an attempt to save  into a new group, also ensure the user has access to the target group.
+            // If it's an attempt to save into a new group, also ensure the user has access to the target group.
             if (scope.targetGroupId && scope.targetGroupId !== scope.currentGroupId) {
               const tMember = targetGroupCount > 0;
 
-              debug(`user ${userId} ${tMember ? 'is a' : 'is not a'} ${roleName} of group ${scope.targetGroupId}`);
-              res = res && tMember;
+              winston.debug(`ROLE RESOLVER - Attempting save into new target group, User ${userId} ${tMember ? 'has role' : 'does not have'} ${roleName} in target group ${scope.targetGroupId}`);
+              bAllowAccess = bAllowAccess && tMember;
             }
           }
 
           // Note the fact that we are allowing access due to passing an ACL.
-          if (res) {
-            winston.info('Allowing access due to passing an ACL');
+           if (bAllowAccess) {
```

Comment:
nice job using more descriptive names for variables 👍 
nitpick here but looks like there's an extra space in front of the `if`